### PR TITLE
53 handle api requests of 5000 objects in move data from api to database

### PR DESCRIPTION
**Pull request by GregoryHo88**

### CODE REVIEW

File: civiclens/collect/access_api_data.py [hunk 1]
```diff
@@ -167,20 +193,67 @@ def poll_for_response(api_key, wait_for_rate_reset):
...
+            if success or (_is_duplicated_on_server(r_json) and skip_duplicates):
+                if doc_data is not None:
+                    doc_data += r_json["data"]
+                else:
+                    doc_data = r_json["data"]
 
-            # Break if it's the last page
-            if r_json["meta"]["lastPage"]:
-                return doc_data
+                # Break if it's the last page
```
Comment (andrewjtdunn) [inline civiclens/collect/access_api_data.py:255]: I don't think this code should change indentation (?)

File: civiclens/collect/access_api_data.py [hunk 2]
```diff
@@ -64,6 +64,29 @@ def api_date_format_params(data_type, start_date=None, end_date=None):
 
     return date_param
 
+def format_datetime_for_api(dt_str):
+    """
+    Converts a UTC datetime string from the API to a formatted string representing Eastern Time.
+
+    This helped function was constructed to process the `lastModifiedDate` timestamp obtained from 
+    the API's response into Eastern Time, that is required when making API requests.
+    Ref: https://open.gsa.gov/api/regulationsgov/#searching-for-comments-1
```
Comment (andrewjtdunn) [inline civiclens/collect/access_api_data.py:73]: just noting this url doesn't load for me

File: civiclens/tests/test_access_api_data.py [hunk 3]
```diff
@@ -47,3 +47,7 @@ def test_pull_reg_gov_data():
         "documents",
         params={"filter[searchTerm]": "CRC-2024-0073-0001"},
     )
+
+def test_format_datetime_for_api():
+    assert access_api_data.format_datetime_for_api(
+        "2020-08-10T15:58:52Z") == "2020-08-10 11:58:52"
```
Comment (andrewjtdunn) [inline civiclens/tests/test_access_api_data.py:53]: do you want to add a test to assert that an improper date format results in an error?

**Review by andrewjtdunn — COMMENTED** [review]
added  a couple comments -- the only big one is the one about indentation

File: civiclens/collect/access_api_data.py (hunk 1 above)
Comment (GregoryHo88) [reply civiclens/collect/access_api_data.py:255]: Hi @andrewjtdunn, when you have a moment, could we schedule a call to discuss this? I'm not sure what you mean, and I'd appreciate your clarification.

File: civiclens/tests/test_access_api_data.py (hunk 3 above)
Comment (GregoryHo88) [reply civiclens/tests/test_access_api_data.py:53]: Adding as an issue to backlog. TBD next week

**Comment by GregoryHo88** [comment]
Hey @andrewjtdunn , Code rectified as discussed.

**Review by andrewjtdunn — APPROVED** [review]
looks good

**Pull request merged**